src/linear_algebra.jl

```diff
@@ -275,28 +274,58 @@ end
 
 # This is incorrect if `C` is `LinearAlgebra.Symmetric` as we modify twice the
 # same diagonal element.
-function _add_mul_array(C::Matrix, A::AbstractMatrix, B::AbstractMatrix)
-    mul_buffer = buffer_for(add_mul, eltype(C), eltype(A), eltype(B))
+function _add_mul_array(buffer, C::Matrix, A::AbstractMatrix, B::AbstractMatrix)
     @inbounds begin
         for i = 1:size(A, 1), j = 1:size(B, 2)
             Ctmp = C[i, j]
             for k = 1:size(A, 2)
-                Ctmp = buffered_operate!(mul_buffer, add_mul, Ctmp, A[i, k], B[k, j])
+                Ctmp = buffered_operate!(buffer, add_mul, Ctmp, A[i, k], B[k, j])
             end
             C[i, j] = Ctmp
         end
     end # @inbounds
     return C
 end
 
-function mutable_operate!(
+function mutable_buffered_operate!(
+    buffer,
+    ::typeof(add_mul),
+    C::VecOrMat,
+    A::AbstractMatrix,
+    B::AbstractVecOrMat,
+)
+    _dim_check(C, A, B)
+    _add_mul_array(buffer, C, A, B)
+end
+# Break ambiguity
```

Comment:
Which ambiguity? It seems weird to have to define the `Nothing` version when it is identical.

Reply:
With
https://github.com/jump-dev/MutableArithmetics.jl/blob/master/src/interface.jl#L289-L291
We could make a fallback to avoid ambiguity